pkg/policy/api/rule_validation.go

```diff
@@ -81,8 +98,29 @@ func (i *IngressRule) sanitize() error {
 }
 
 func (e *EgressRule) sanitize() error {
-	if len(e.ToCIDR) > 0 && len(e.ToPorts) > 0 {
-		return fmt.Errorf("Combining ToPorts and ToCIDR is not supported yet")
+	l3Members := map[string]int{
+		"ToCIDR":     len(e.ToCIDR),
+		"ToCIDRSet":  len(e.ToCIDRSet),
+		"ToEntities": len(e.ToEntities),
+		"ToServices": len(e.ToServices),
+	}
+	l3DependentL4Support := map[interface{}]bool{
+		"ToCIDR":     false,
+		"ToCIDRSet":  false,
+		"ToEntities": false,
+		"ToServices": false,
+	}
+	for m1 := range l3Members {
+		for m2 := range l3Members {
+			if m2 != m1 && l3Members[m1] > 0 && l3Members[m2] > 0 {
+				return fmt.Errorf("Combining %s and %s is not supported yet", m1, m2)
+			}
+		}
+	}
+	for member := range l3Members {
+		if l3Members[member] > 0 && len(e.ToPorts) > 0 && !l3DependentL4Support[member] {
```

Comment:
Do we still need the `!l3DependentL4Support[member]` check in egress, since all values are false?

Reply:
Not strictly, no, but egress label-dependent l4 is pretty close so it seems easy enough to just include this here. It'll require a minor rebase and update for the egress work.